docs/import/manage_disaster.js

```diff
@@ -635,34 +637,67 @@ function writeNewDisaster(disasterId, states) {
     setStatus('Error: disaster with that name and year already exists.');
     return Promise.resolve(false);
   }
+  clearStatus();
   const currentData = createDisasterData(states);
   disasterData.set(disasterId, currentData);
 
-  const disasterPicker = $('#disaster-dropdown');
-  const disasterOptions = disasterPicker.children();
-  let added = false;
-  // We expect this recently created disaster to go near the top of the list, so
-  // do a linear scan down.
-  // Note: let's hope this tool isn't being used in the year 10000.
-  // Comment needed to quiet eslint.
-  disasterOptions.each(/* @this HTMLElement */ function() {
-    if ($(this).val() < disasterId) {
-      $(createOptionFrom(disasterId)).insertBefore($(this));
-      added = true;
-      return false;
-    }
-  });
-  if (!added) disasterPicker.append(createOptionFrom(disasterId));
-  toggleState(true);
+  const folderCreationPromises = [];
+  folderCreationPromises.push(
+      getCreateFolderPromise(eeLegacyPathPrefix + disasterId));
+  for (const state of states) {
+    folderCreationPromises.push(
+        getCreateFolderPromise(legacyStateDir + '/' + state));
+  }
+
+  Promise.all(folderCreationPromises).then(() => {
+    const disasterPicker = $('#disaster-dropdown');
+    const disasterOptions = disasterPicker.children();
+    let added = false;
+    // We expect this recently created disaster to go near the top of the list,
+    // so do a linear scan down. Note: let's hope this tool isn't being used in
+    // the year 10000. Comment needed to quiet eslint.
+    disasterOptions.each(/* @this HTMLElement */ function() {
+      if ($(this).val() < disasterId) {
+        $(createOptionFrom(disasterId)).insertBefore($(this));
+        added = true;
+        return false;
+      }
+    });
+    if (!added) disasterPicker.append(createOptionFrom(disasterId));
+    toggleState(true);
 
-  disasterPicker.val(disasterId).trigger('change');
+    disasterPicker.val(disasterId).trigger('change');
+  });
 
   return disasterCollectionReference()
       .doc(disasterId)
       .set(currentData)
       .then(() => true);
 }
 
+/**
+ * Returns a promise that resolves on the creation of the given folder.
+ *
+ * This will print a console error for anyone other than the gd
+ * account. Ee console seems to have the power to grant write access
+ * to non-owners but it doesn't seem to work. Sent an email to
+ * gestalt.
+ * TODO: replace with setIamPolicy when that works.
+ * TODO: add status bar for when this is finished.
+ *
+ * @param {string} dir asset path of folder to create
+ * @return {Promise<void>} resolves when after the directory is created and
+ * set to world readable.
+ */
+function getCreateFolderPromise(dir) {
+  return new Promise((resolve) => {
+    ee.data.createFolder(dir, false, () => {
+      ee.data.setAssetAcl(dir, {all_users_can_read: true});
```

Comment:
I think maybe we should make this async as well? Then can just pass resolve as the callback, and one-line everything :)